sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -52,5 +64,90 @@ public void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position)
             Policies ??= new();
             Policies.Add((position, policy));
         }
+
+        /// <summary>
+        /// Adds a custom classifier to the <see cref="ResponseClassifier"/> that decides if the response
+        /// received from the service should be considered an error response, for this service call.
+        /// The custom classifier is applied before the default classifier.
+        /// This is useful in cases where you'd like to prevent specific response status codes from appearing as errors in
+        /// logging and distributed tracing.  It will also prevent the call from throwing an exception when a response with
+        /// this status code is received.
+        /// </summary>
+        /// <param name="statusCode">The status code to customize classification for.</param>
+        /// <param name="isError">Whether the passed-in status code should be classified as an error.</param>
+        public void AddClassifier(int statusCode, bool isError)
```

Comment:
Are we concerned about thead-safety for this and the other overload to `AddClassifer`?  I believe competing callers could result in us dropping a classifer.

Reply:
I've started a thread with the DPG design team, and created https://github.com/azure/azure-sdk-for-net/issues/27007 to track the issue.